internal/adapters/cloud/aws/rds/rds.go

```diff
@@ -50,9 +50,41 @@ func (a *adapter) Adapt(root *aws2.RootAdapter, state *state.State) error {
 		return nil
 	}
 
+	state.AWS.RDS.Snapshots, err = a.getSnapshots()
+	if err != nil {
+		return err
+	}
+
+	state.AWS.RDS.ParameterGroups, err = a.getParameterGroups()
+	if err != nil {
+		return err
+	}
+
 	return nil
 }
 
+func (a *adapter) getSnapshots() (snapshots []rds.Snapshots, err error) {
+	a.Tracker().SetServiceLabel("Discovering Snapshots...")
+	var apiDBSnapshots []types.DBSnapshot
+	var input rdsApi.DescribeDBSnapshotsInput
+
+	for {
+		output, err := a.api.DescribeDBSnapshots(a.Context(), &input)
+		if err != nil {
+			return nil, err
+		}
+
+		apiDBSnapshots = append(apiDBSnapshots, output.DBSnapshots...)
+		a.Tracker().SetTotalResources(len(apiDBSnapshots))
+		if output.Marker == nil {
+			break
+		}
+		input.Marker = output.Marker
+	}
```

Comment:
What if neither if the break out conditions are ever true? This loop will never end.

Can we find another way to terminate the loop before we get into such a situation?